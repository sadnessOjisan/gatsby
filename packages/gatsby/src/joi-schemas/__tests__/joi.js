const { gatsbyConfigSchema } = require(`../joi`)

describe(`gatsby config`, () => {
  it(`strips trailing slashes from url fields`, () => {
    const config = {
      pathPrefix: `/blog///`,
      assetPrefix: `https://cdn.example.com/`,
    }

    expect(gatsbyConfigSchema.validate(config)).resolves.toEqual({
      pathPrefix: `/blog`,
      assetPrefix: `https://cdn.example.com`,
    })
  })

  it(`allows assetPrefix to be full URL`, () => {
    const config = {
      assetPrefix: `https://cdn.example.com`,
    }

    expect(gatsbyConfigSchema.validate(config)).resolves.toEqual(config)
  })

  it(`allows assetPrefix to be a URL with nested paths`, () => {
    const config = {
      assetPrefix: `https://cdn.example.com/some/nested/path`,
    }

    expect(gatsbyConfigSchema.validate(config)).resolves.toEqual(config)
  })

  it(`allows relative paths for url fields`, () => {
    const config = {
      pathPrefix: `/blog`,
      assetPrefix: `https://cdn.example.com`,
    }

    expect(gatsbyConfigSchema.validate(config)).resolves.toEqual(config)
  })

  it(`does not allow pathPrefix to be full URL`, () => {
    const config = {
      pathPrefix: `https://google.com`,
    }

    expect(
      gatsbyConfigSchema.validate(config)
    ).rejects.toThrowErrorMatchingSnapshot()
  })

  it(`throws when relative path used for both assetPrefix and pathPrefix`, () => {
    const config = {
      assetPrefix: `/assets`,
      pathPrefix: `/blog`,
    }

    expect(
      gatsbyConfigSchema.validate(config)
    ).rejects.toThrowErrorMatchingSnapshot()
  })
})
