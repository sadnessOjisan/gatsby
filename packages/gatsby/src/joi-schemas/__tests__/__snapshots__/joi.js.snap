// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gatsby config does not allow pathPrefix to be full URL 1`] = `"child \\"pathPrefix\\" fails because [\\"pathPrefix\\" must be a valid relative uri]"`;

exports[`gatsby config throws when relative path used for both assetPrefix and pathPrefix 1`] = `"assetPrefix must be an absolute URI when used with pathPrefix"`;
